spec/ruby/core/array/sample_spec.rb

```diff
@@ -2,6 +2,13 @@
 require File.expand_path('../fixtures/classes', __FILE__)
 
 describe "Array#sample" do
+  it "samples evenly" do
+    3.times do |i|
+      samples = 4000.times.map { 4.times.to_a.sample(3)[i] }
+      counts = samples.group_by(&:itself).map { |a, b| b.size }
+      counts.each { |count| (800..1200).should include(count) }
```

Comment:
How I choose 800..1200:
probability of count in 800..1200 is 0.9999999999996083

if you change this to 900..1100, probability will be 0.99976
it means that the test wil fail in 0.024% even if there is no bug.

```ruby
def overflow_safe_mult values
  a = 1
  values = values.sort_by(&:abs)
  a *= a.abs>1 ? values.shift : values.pop until values.empty?
  a
end
def probability_of_count n, p, range
  range.map{|i|
    overflow_safe_mult [p]*i+[1-p]*(n-i)+i.times.map{|j|n-j}+i.times.map{|j|1.0/(j+1)}
  }.inject(:+)
end
probability_of_count 4000, 0.25, 0..799 #=> 3.181654463145792e-14
probability_of_count 4000, 0.25, 800..1200 #=> 0.9999999999996083
probability_of_count 4000, 0.25, 1201..4000 #=> 3.572627145186968e-13

probability_of_count 4000, 0.25, 900..1100 #=> 0.999756654675964
```